$OpenBSD: patch-configure,v 1.12 2013/04/13 09:11:02 landry Exp $

- Chunk 1, 2 & 6 : https://bugs.webkit.org/show_bug.cgi?id=109932
Only webkit2 requires c++11 features atm, so let's relax the check until we have webkit2

- Chunk 3 : GNOME Bug 627126 - gsettings schema files don't get installed on FreeBSD
glib commit 136e705e8383ff8848e425ac01278102d7badc52

- Chunk 4 & 5 & 8 : fix bashisms
https://bugs.webkit.org/show_bug.cgi?id=113349

- Chunk 7 : Don't force the use of -O2, the ports infrastructure
takes care of that

- Chunk 9 : use MODPY_BIN when calling the generator
--- configure.orig	Tue Mar 26 20:46:03 2013
+++ configure	Thu Apr 11 19:46:57 2013
@@ -4717,7 +4717,7 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
-#if !(defined(__GNUC__) && !defined(__clang__) && !defined(__INTEL_COMPILER) && __GNUC__ >= 4 && __GNUC_MINOR__ >= 7) \
+#if !(defined(__GNUC__) && !defined(__clang__) && !defined(__INTEL_COMPILER) && __GNUC__ >= 4 && __GNUC_MINOR__ >= 2) \
     && !(defined(__clang__) && __clang_major__ >= 3 && __clang_minor__ >= 0)
 #error Unsupported compiler
 #endif
@@ -4754,7 +4754,7 @@ ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
-#if !(defined(__GNUG__) && defined(__GNUC__) && !defined(__clang__) && !defined(__INTEL_COMPILER) && __GNUC__ >= 4 && __GNUC_MINOR__ >= 7) \
+#if !(defined(__GNUG__) && defined(__GNUC__) && !defined(__clang__) && !defined(__INTEL_COMPILER) && __GNUC__ >= 4 && __GNUC_MINOR__ >= 2) \
     && !(defined(__clang__) && __clang_major__ >= 3 && __clang_minor__ >= 0)
 #error Unsupported compiler
 #endif
@@ -17479,26 +17479,14 @@ gsettings__enum_file = $(addsuffix .enums.xml,$(gsetti
 
 all-am: $(gsettings_SCHEMAS:.xml=.valid)
 uninstall-am: uninstall-gsettings-schemas
-install-data-am: install-gsettings-schemas
+install-data-am:
 
 .SECONDARY: $(gsettings_SCHEMAS)
 
-gsettings__base_list = \
-  sed "$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;s/\n/ /g" | \
-  sed "$$!N;$$!N;$$!N;$$!N;s/\n/ /g"
-
-install-gsettings-schemas: $(gsettings_SCHEMAS:.xml=.valid) $(gsettings__enum_file)
+install-gsettings-schemas: $(gsettings_SCHEMAS) $(gsettings__enum_file)
 	@$(NORMAL_INSTALL)
 	test -z "$(gsettingsschemadir)" || $(MKDIR_P) "$(DESTDIR)$(gsettingsschemadir)"
-	@list='\''$(gsettings__enum_file) $(gsettings_SCHEMAS)'\''; test -n "$(gsettingsschemadir)" || list=; \
-	for p in $$list; do \
-	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
-	  echo "$$d$$p"; \
-	done | $(gsettings__base_list) | \
-	while read files; do \
-	  echo " $(INSTALL_DATA) $$files '\''$(DESTDIR)$(gsettingsschemadir)'\''"; \
-	  $(INSTALL_DATA) $$files "$(DESTDIR)$(gsettingsschemadir)" || exit $$?; \
-	done
+	$(INSTALL_DATA) $^ "$(DESTDIR)$(gsettingsschemadir)"
 	test -n "$(GSETTINGS_DISABLE_SCHEMAS_COMPILE)$(DESTDIR)" || $(GLIB_COMPILE_SCHEMAS) $(gsettingsschemadir)
 
 uninstall-gsettings-schemas:
@@ -20218,18 +20206,18 @@ fi
 
 if test "$with_acceleration_backend" = "opengl"; then
     if test "$enable_gles2" = "yes"; then
-        acceleration_backend_description+= "(gles2"
+        acceleration_backend_description= "(gles2"
         OPENGL_LIBS="-lGLESv2"
     else
-        acceleration_backend_description+="(gl"
+        acceleration_backend_description="(gl"
         OPENGL_LIBS="-lGL"
     fi
     if test "$enable_egl" = "yes"; then
-        acceleration_backend_description+=", egl"
-        OPENGL_LIBS+=" -lEGL"
+        acceleration_backend_description="$acceleration_backend_description, egl"
+        OPENGL_LIBS="$OPENGL_LIBS -lEGL"
     fi
     if test "$enable_glx" = "yes"; then
-        acceleration_backend_description+=", glx"
+        acceleration_backend_description="$acceleration_backend_description, glx"
     fi
 
     # Check whether dlopen() is in the core libc like on FreeBSD, or in a separate
@@ -20282,8 +20270,8 @@ fi
 
 
 
-    OPENGL_LIBS+=" $DLOPEN_LIBS"
-    acceleration_backend_description+=")"
+    OPENGL_LIBS="$OPENGL_LIBS $DLOPEN_LIBS"
+    acceleration_backend_description="$acceleration_backend_description)"
 fi
 
 
@@ -21037,7 +21025,7 @@ fi
 # Use C99 as the language standard for C code.
 CFLAGS="$CFLAGS -std=c99"
 # Do not warn about C++11 incompatibilities.
-CXXFLAGS="$CXXFLAGS -Wno-c++11-compat"
+# CXXFLAGS="$CXXFLAGS -Wno-c++11-compat"
 
 # Clang requires suppression of unused arguments warning.
 if test "$CC" = "clang"; then
@@ -21065,8 +21053,8 @@ fi
 
 # Add the appropriate 'O' level for optimized builds.
 if test "$enable_optimizations" = "yes"; then
-    CXXFLAGS="$CXXFLAGS -O2"
-    CFLAGS="$CFLAGS -O2"
+    CXXFLAGS="$CXXFLAGS"
+    CFLAGS="$CFLAGS"
 else
     CXXFLAGS="$CXXFLAGS -O0"
     CFLAGS="$CFLAGS -O0"
@@ -21336,100 +21324,100 @@ fi
 # This list of features represents those selected for release builds.
 # If you are adding a new or unstable feature, you should mark it
 # disabled here.
-read -d '' DEFAULT_FEATURE_DEFINES <<"EOF"
-	ENABLE_ACCELERATED_2D_CANVAS=0
-	ENABLE_BATTERY_STATUS=0
-	ENABLE_BLOB=1
-	ENABLE_CANVAS_PATH=0
-	ENABLE_CANVAS_PROXY=0
-	ENABLE_CHANNEL_MESSAGING=1
-	ENABLE_CSP_NEXT=0
-	ENABLE_CSS3_TEXT=0
-	ENABLE_CSS_BOX_DECORATION_BREAK=1
-	ENABLE_CSS_COMPOSITING=0
-	ENABLE_CSS_DEVICE_ADAPTATION=0
-	ENABLE_CSS_EXCLUSIONS=1
-	ENABLE_CSS_IMAGE_ORIENTATION=0
-	ENABLE_CSS_IMAGE_RESOLUTION=0
-	ENABLE_CSS_IMAGE_SET=0
-	ENABLE_CSS_REGIONS=0
-	ENABLE_CSS_STICKY_POSITION=0
-	ENABLE_CSS_TRANSFORMS_ANIMATIONS_UNPREFIXED=0
-	ENABLE_CSS_VARIABLES=0
-	ENABLE_CSS3_CONDITIONAL_RULES=0
-	ENABLE_CUSTOM_SCHEME_HANDLER=0
-	ENABLE_DASHBOARD_SUPPORT=0
-	ENABLE_DATALIST_ELEMENT=0
-	ENABLE_DATA_TRANSFER_ITEMS=0
-	ENABLE_DETAILS_ELEMENT=1
-	ENABLE_DEVICE_ORIENTATION=0
-	ENABLE_DIALOG_ELEMENT=0
-	ENABLE_DIRECTORY_UPLOAD=0
-	ENABLE_DOM4_EVENTS_CONSTRUCTOR=0
-	ENABLE_DOWNLOAD_ATTRIBUTE=0
-	ENABLE_FILE_SYSTEM=0
-	ENABLE_FILTERS=1
-	ENABLE_FTPDIR=0
-	ENABLE_FULLSCREEN_API=1
-	ENABLE_HIGH_DPI_CANVAS=0
-	ENABLE_ICONDATABASE=1
-	ENABLE_IFRAME_SEAMLESS=1
-	ENABLE_INDEXED_DATABASE=0
-	ENABLE_INPUT_SPEECH=0
-	ENABLE_INPUT_TYPE_COLOR=0
-	ENABLE_INPUT_TYPE_DATE=0
-	ENABLE_INPUT_TYPE_DATETIME=0
-	ENABLE_INPUT_TYPE_DATETIMELOCAL=0
-	ENABLE_INPUT_TYPE_MONTH=0
-	ENABLE_INPUT_TYPE_TIME=0
-	ENABLE_INPUT_TYPE_WEEK=0
-	ENABLE_INSPECTOR=1
-	ENABLE_JAVASCRIPT_DEBUGGER=1
-	ENABLE_LEGACY_CSS_VENDOR_PREFIXES=0
-	ENABLE_LEGACY_NOTIFICATIONS=0
-	ENABLE_LEGACY_VENDOR_PREFIXES=1
-	ENABLE_LEGACY_WEB_AUDIO=0
-	ENABLE_LINK_PREFETCH=0
-	ENABLE_LINK_PRERENDER=0
-	ENABLE_MATHML=0
-	ENABLE_MEDIA_CAPTURE=0
-	ENABLE_MEDIA_SOURCE=0
-	ENABLE_MEDIA_STATISTICS=0
-	ENABLE_MEDIA_STREAM=0
-	ENABLE_METER_ELEMENT=1
-	ENABLE_MHTML=1
-	ENABLE_MICRODATA=0
-	ENABLE_MOUSE_CURSOR_SCALE=0
-	ENABLE_NAVIGATOR_CONTENT_UTILS=0
-	ENABLE_NETSCAPE_PLUGIN_API=1
-	ENABLE_NETWORK_INFO=0
-	ENABLE_NOTIFICATIONS=0
-	ENABLE_ORIENTATION_EVENTS=0
-	ENABLE_PAGE_VISIBILITY_API=0
-	ENABLE_PERFORMANCE_TIMELINE=0
-	ENABLE_PROGRESS_ELEMENT=1
-	ENABLE_PROXIMITY_EVENTS=0
-	ENABLE_QUOTA=0
-	ENABLE_RESOURCE_TIMING=0
-	ENABLE_REQUEST_ANIMATION_FRAME=1
-	ENABLE_SCRIPTED_SPEECH=0
-	ENABLE_SHADOW_DOM=1
-	ENABLE_SHARED_WORKERS=1
-	ENABLE_SMOOTH_SCROLLING=1
-	ENABLE_SQL_DATABASE=1
-	ENABLE_STYLE_SCOPED=0
-	ENABLE_SVG_DOM_OBJC_BINDINGS=0
-	ENABLE_TEMPLATE_ELEMENT=0
-	ENABLE_TEXT_AUTOSIZING=0
-	ENABLE_TOUCH_EVENTS=0
-	ENABLE_TOUCH_ICON_LOADING=0
-	ENABLE_USER_TIMING=0
-	ENABLE_VIBRATION=0
-	ENABLE_VIDEO_TRACK=0
-	ENABLE_WEB_AUDIO=0
-	ENABLE_WEB_SOCKETS=1
-	ENABLE_WEB_TIMING=1
-	ENABLE_WORKERS=1
+read DEFAULT_FEATURE_DEFINES <<"EOF"
+	ENABLE_ACCELERATED_2D_CANVAS=0 \
+	ENABLE_BATTERY_STATUS=0 \
+	ENABLE_BLOB=1 \
+	ENABLE_CANVAS_PATH=0 \
+	ENABLE_CANVAS_PROXY=0 \
+	ENABLE_CHANNEL_MESSAGING=1 \
+	ENABLE_CSP_NEXT=0 \
+	ENABLE_CSS3_TEXT=0 \
+	ENABLE_CSS_BOX_DECORATION_BREAK=1 \
+	ENABLE_CSS_COMPOSITING=0 \
+	ENABLE_CSS_DEVICE_ADAPTATION=0 \
+	ENABLE_CSS_EXCLUSIONS=1 \
+	ENABLE_CSS_IMAGE_ORIENTATION=0 \
+	ENABLE_CSS_IMAGE_RESOLUTION=0 \
+	ENABLE_CSS_IMAGE_SET=0 \
+	ENABLE_CSS_REGIONS=0 \
+	ENABLE_CSS_STICKY_POSITION=0 \
+	ENABLE_CSS_TRANSFORMS_ANIMATIONS_UNPREFIXED=0 \
+	ENABLE_CSS_VARIABLES=0 \
+	ENABLE_CSS3_CONDITIONAL_RULES=0 \
+	ENABLE_CUSTOM_SCHEME_HANDLER=0 \
+	ENABLE_DASHBOARD_SUPPORT=0 \
+	ENABLE_DATALIST_ELEMENT=0 \
+	ENABLE_DATA_TRANSFER_ITEMS=0 \
+	ENABLE_DETAILS_ELEMENT=1 \
+	ENABLE_DEVICE_ORIENTATION=0 \
+	ENABLE_DIALOG_ELEMENT=0 \
+	ENABLE_DIRECTORY_UPLOAD=0 \
+	ENABLE_DOM4_EVENTS_CONSTRUCTOR=0 \
+	ENABLE_DOWNLOAD_ATTRIBUTE=0 \
+	ENABLE_FILE_SYSTEM=0 \
+	ENABLE_FILTERS=1 \
+	ENABLE_FTPDIR=0 \
+	ENABLE_FULLSCREEN_API=1 \
+	ENABLE_HIGH_DPI_CANVAS=0 \
+	ENABLE_ICONDATABASE=1 \
+	ENABLE_IFRAME_SEAMLESS=1 \
+	ENABLE_INDEXED_DATABASE=0 \
+	ENABLE_INPUT_SPEECH=0 \
+	ENABLE_INPUT_TYPE_COLOR=0 \
+	ENABLE_INPUT_TYPE_DATE=0 \
+	ENABLE_INPUT_TYPE_DATETIME=0 \
+	ENABLE_INPUT_TYPE_DATETIMELOCAL=0 \
+	ENABLE_INPUT_TYPE_MONTH=0 \
+	ENABLE_INPUT_TYPE_TIME=0 \
+	ENABLE_INPUT_TYPE_WEEK=0 \
+	ENABLE_INSPECTOR=1 \
+	ENABLE_JAVASCRIPT_DEBUGGER=1 \
+	ENABLE_LEGACY_CSS_VENDOR_PREFIXES=0 \
+	ENABLE_LEGACY_NOTIFICATIONS=0 \
+	ENABLE_LEGACY_VENDOR_PREFIXES=1 \
+	ENABLE_LEGACY_WEB_AUDIO=0 \
+	ENABLE_LINK_PREFETCH=0 \
+	ENABLE_LINK_PRERENDER=0 \
+	ENABLE_MATHML=0 \
+	ENABLE_MEDIA_CAPTURE=0 \
+	ENABLE_MEDIA_SOURCE=0 \
+	ENABLE_MEDIA_STATISTICS=0 \
+	ENABLE_MEDIA_STREAM=0 \
+	ENABLE_METER_ELEMENT=1 \
+	ENABLE_MHTML=1 \
+	ENABLE_MICRODATA=0 \
+	ENABLE_MOUSE_CURSOR_SCALE=0 \
+	ENABLE_NAVIGATOR_CONTENT_UTILS=0 \
+	ENABLE_NETSCAPE_PLUGIN_API=1 \
+	ENABLE_NETWORK_INFO=0 \
+	ENABLE_NOTIFICATIONS=0 \
+	ENABLE_ORIENTATION_EVENTS=0 \
+	ENABLE_PAGE_VISIBILITY_API=0 \
+	ENABLE_PERFORMANCE_TIMELINE=0 \
+	ENABLE_PROGRESS_ELEMENT=1 \
+	ENABLE_PROXIMITY_EVENTS=0 \
+	ENABLE_QUOTA=0 \
+	ENABLE_RESOURCE_TIMING=0 \
+	ENABLE_REQUEST_ANIMATION_FRAME=1 \
+	ENABLE_SCRIPTED_SPEECH=0 \
+	ENABLE_SHADOW_DOM=1 \
+	ENABLE_SHARED_WORKERS=1 \
+	ENABLE_SMOOTH_SCROLLING=1 \
+	ENABLE_SQL_DATABASE=1 \
+	ENABLE_STYLE_SCOPED=0 \
+	ENABLE_SVG_DOM_OBJC_BINDINGS=0 \
+	ENABLE_TEMPLATE_ELEMENT=0 \
+	ENABLE_TEXT_AUTOSIZING=0 \
+	ENABLE_TOUCH_EVENTS=0 \
+	ENABLE_TOUCH_ICON_LOADING=0 \
+	ENABLE_USER_TIMING=0 \
+	ENABLE_VIBRATION=0 \
+	ENABLE_VIDEO_TRACK=0 \
+	ENABLE_WEB_AUDIO=0 \
+	ENABLE_WEB_SOCKETS=1 \
+	ENABLE_WEB_TIMING=1 \
+	ENABLE_WORKERS=1 \
 	ENABLE_XHR_TIMEOUT=1
 EOF
 
@@ -21501,7 +21489,7 @@ fi
 
 # We need to run this command now rather than use AC_CONFIG_COMMANDS
 # because automake rules depend on the output file (WebKitFeatures.txt).
-$srcdir/Tools/gtk/generate-feature-defines-files $DEFAULT_FEATURE_DEFINES
+${MODPY_BIN} $srcdir/Tools/gtk/generate-feature-defines-files $DEFAULT_FEATURE_DEFINES
 
 
 
